### PR TITLE
fix gateway timeout to controller 

### PR DESCRIPTION
**Pull request by rabbah**
Fix gateway timeout to controller - wait up 5 minutes which is max allowed action duration.

### CODE REVIEW

File: ansible/roles/nginx/templates/nginx.conf.j2 [hunk 1]
```diff
@@ -37,6 +37,7 @@ http {
         }
         location /api/v1 {
             proxy_pass http://{{ groups['controllers']|first }}:{{ controller.port }};
+            proxy_read_timeout 300s;
```
Comment (rabbah) [inline ansible/roles/nginx/templates/nginx.conf.j2]: default is 60s.

**Comment by rabbah** [comment]
@lehoanganh @domdom82 FYI

File: ansible/roles/nginx/templates/nginx.conf.j2 (hunk 1 above)
Comment (markusthoemmes) [inline ansible/roles/nginx/templates/nginx.conf.j2]: Do we need little bit of slack, as the action, even when called blocking will return in 5 minutes + the time needed to propagate stuff?

Comment (rabbah) [inline ansible/roles/nginx/templates/nginx.conf.j2]: 305s?

Comment (markusthoemmes) [inline ansible/roles/nginx/templates/nginx.conf.j2]: Something along these lines, 305 should be enough, 310 to be sure?

Comment (rabbah) [inline ansible/roles/nginx/templates/nginx.conf.j2]: btw controller grants no more than 5s grace.

Comment (markusthoemmes) [inline ansible/roles/nginx/templates/nginx.conf.j2]: So it's going to be 305 (including grace) + some nanoseconds at least to get the request back to nginx so it doesn't say: Gateway Timeout?

**Comment by rabbah** [comment]
rebased.

**Comment by sjfink** [comment]
rebased and merged

**Pull request closed without merging**